include/evmc/helpers.h

```diff
@@ -106,6 +108,55 @@ static inline struct evmc_result evmc_execute(struct evmc_instance* instance,
     return instance->execute(instance, context, rev, msg, code, code_size);
 }
 
+/// The evmc_result release function using free() for releasing the memory.
+///
+/// This function is used in the evmc_make_result(),
+/// but may be also used in other case if convenient.
+///
+/// @param result The result object.
+static void evmc_free_result_memory(const struct evmc_result* result)
+{
+    free((uint8_t*)result->output_data);
+}
+
+/// Creates the result from the provided arguments.
+///
+/// The provided output is copied to memory allocated with malloc()
+/// and the evmc_result::release function is set to one invoking free().
+///
+/// @param status_code  The status code.
+/// @param gas_left     The amount of gas left.
+/// @param output_data  The pointer to the output.
+/// @param output_size  The output size.
+static inline struct evmc_result evmc_make_result(enum evmc_status_code status_code,
+                                                  int64_t gas_left,
+                                                  const uint8_t* output_data,
+                                                  size_t output_size)
+{
+    struct evmc_result result;
+    memset(&result, 0, sizeof(result));
+
+    if (output_size != 0)
+    {
+        uint8_t* buffer = (uint8_t*)malloc(output_size);
+
+        if (!buffer)
+        {
+            result.status_code = EVMC_OUT_OF_MEMORY;
```

Comment:
Still not convinced this it the right way to do it, given it confuses status of the execution vs. helpers, but in the ned if OOM is hit there is a bigger underlying issue. Perhaps worth highlighting this in the description above?

Reply:
> but if OOM is hit there is a bigger underlying issue

What?

I don't thing this is the best way of passing output buffer. This will rather be only useful for Go, so we can drop this PR.

Reply:
> > but if OOM is hit there is a bigger underlying issue
>
> What?

OOM = out of memory

Basically I'm ok with this change given this new specific error code is introduced.

Reply:
No, I had issue with understanding the sentence after "OOM". The OOM itself is difficult to handle in general. E.g. it will never happen on Linux in practice as it will happily try to allocate 2 TB of memory and in the end kill the process.

Anyway, the new specific error code was introduced: https://github.com/ethereum/evmc/pull/333/files#diff-8e9caa43721b131d9f7897e2bfd98299R309.

Reply:
Yeah I know, that's the line I commented on. Just having this documented on `make_result` would be useful.